Settings screen refactor +  Calendar weekdays selection

NecroKote opened the pull request:
This PR contains:
- fix for #234 via refactoring of Settings screen. Rows are now bundled as `SectionHeader` and `SettingRow`'s making `SettingScreen` way more readable and uniform. Also I've altered typography styles to make the page bit easier on the eyes
- fix for #267 by adding a row with days of the week to the calendar. Saturdays and Sundays are slightly "highlighted"
- style improvement for calendar selection boundries

Check attached screenshots:
<p align="center" float="left">
<img src="https://github.com/user-attachments/assets/735650d6-d11b-4898-a34c-cefa4b1c2b0b" width="150px"/>
<img alt="day of the week row" src="https://github.com/user-attachments/assets/78c8765c-c510-4c25-96d1-42f9ed2b2a7f" width="150px"/>
<img alt="calendar selection style" src="https://github.com/user-attachments/assets/72052dea-f616-4386-a818-880ef401fae7" width="150px"/>
</p>

Commit:
refactor `SettingScreen` unifying look of setting rows

Commit:
add day of the week display to calendar

Commit:
fix calendar selection shape

- round when it's a one-day selection
- square if it's a day in between
- has rounded corners if it's on the edges of selection

CrazyMarvin commented:
Thanks again, @NecroKote! 👍 

I like the original date selection in the calendar.
Here is the [mockup](https://www.figma.com/file/9Fk92SQVItIwAjVfsakopo/) in case you did not see it yet.
It is not implemented 1:1 at the moment though. Can you fix it?

Can you make showing the days of the week optional please?

The other changes look good.

Commit:
bring selected range closer to mock-up

Commit:
make show weekdays header optional

NecroKote commented:
@CrazyMarvin always welcome 🙇 

I made the header optional and adjusted the selection a bit so that it's closer to the mockup. Not 100% there yet due to 1.2 aspect ratio preventing the proper circle, but that's something I might take a look later

(ps: updated screenshots)

CrazyMarvin commented:
> @CrazyMarvin always welcome 🙇 
> 
> I made the header optional and adjusted the selection a bit so that it's closer to the mockup. Not 100% there yet due to 1.2 aspect ratio preventing the proper circle, but that's something I might take a look later
> 
> (ps: updated screenshots)

Nice. 😃

By the way, why is the same icon in your screenshot for backup and restore? There should be two different ones.

NecroKote commented:
Aww man ( 
Good old ctrl-c ctrl-v. Will fix tomorrow

Commit:
fix db restore icon
